source/code/providers/support/runasprovider.cpp

```diff
@@ -37,6 +38,14 @@ namespace SCXCore
             }
 
             ParseConfiguration();
+            // set tmpdir location when RunAs provider gets loaded. This will later be used in 
+            // every ExecuteScript call.
+            if ( NULL == sysInfo )
+            {
+            	SCX_LOGINFO(m_log, L"Setting Temp Dir location to: " + s_defaultTmpDir);
+            	sysInfo = SCXCoreLib::SCXHandle<SCXSystemLib::SystemInfo> (new SCXSystemLib::SystemInfo());
+            	sysInfo->SetDefaultTemporaryDirectory(s_defaultTmpDir);
+            }
```

Comment:
See comments elsewhere. Just insure that /etc/opt/microsoft/scx/conf/tmpdir/ exists (if not, then use /tmp/ instead). If it exists, then use it to point to the temporary directory.

You don't need to store a SystemInfo project to do this. Just store it yourself here.